library/Vanilla/EmbeddedContent/Factories/PanoptoEmbedFactory.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\EmbeddedContent\Factories;
+
+use Vanilla\EmbeddedContent\AbstractEmbed;
+use Vanilla\EmbeddedContent\AbstractEmbedFactory;
+use Vanilla\EmbeddedContent\Embeds\PanoptoEmbed;
+
+/**
+ * PanoptoEmbedFactory class.
+ */
+class PanoptoEmbedFactory extends AbstractEmbedFactory {
+
+    /** @var int DEFAULT_HEIGHT */
+    private const DEFAULT_HEIGHT = 360;
+
+    /** @var int DEFAULT_WIDTH */
+    private const DEFAULT_WIDTH = 640;
+
+    /** @var array DOMAINS */
+    private const DOMAINS = [
+        'hosted.panopto.com',
+        'ca.panopto.com',
+        'cloud.panopto.eu',
+        'ap.panopto.com',
+    ];
+
+    /**
+     * Use the page scraper to scrape page data.
+     *
+     * @inheritdoc
+     */
+    public function createEmbedForUrl(string $url): AbstractEmbed {
+        $path = parse_url($url, PHP_URL_HOST);
+
+        return new PanoptoEmbed([
+            'embedType' => PanoptoEmbed::TYPE,
+            'domain' => $path,
+            'url' => $url,
+            'height' => self::DEFAULT_HEIGHT,
+            'width' => self::DEFAULT_WIDTH,
```

Comment:
I am not sure if the embed URL returns frame ratio.
But in our other embed factory implementations we have it set up that way:
```
            'height' => $response['height'] ?? null,
            'width' => $response['width'] ?? null,

Reply:
I saw in other embed factories that we ofter send requests to some API to retrieve information around the embed. However, in this particular case, all the information we need to build the embed is part of the URL, but not these two values.

The embed is generated in Javascript and the height and width can be passed when instantiating an EmbedApi object.

I created these two constants forcing the height and width to respect the ratio of the videos.

Does that make sense?

Reply:
Partially, what will happen if we are in mobile mode?

Reply:
Good catch, I removed the width and height constants declared in the factory and added the proper classes on the embed itself.